src/deploy/firestore/prepare.ts

```diff
@@ -7,6 +7,17 @@ import utils = require("../../utils");
 import { Options } from "../../options";
 import * as fsConfig from "../../firestore/fsConfig";
 
+export interface RulesContext {
+  databaseId: string;
+  rulesFile: string;
+}
+
+export interface IndexContext {
+  databaseId: string;
+  indexesFileName: string;
+  indexesSrc: any;
```

Comment:
What is `indexesSrc`? Is there a more-specific type you can give it than any?

Reply:
indexesSrc is an object describing the raw index source spec found in the config.

the issue here is this may represent the old v1beta1 indexes spec or the new v1/v1 format. The FirestoreIndexes.deploy function does the messy work of converting it from `any` into `Spec.IndexFile` so to minimize changes i think it's acceptable to keep it as `any` here.

i will rename it to `indexesRawSrc` to be more specific